tests/regression/test_ensembleparallelism.py

```diff
@@ -187,6 +187,18 @@ def test_ensemble_reduce(ensemble, mesh, W, urank, urank_sum, root, blocking):
     else:
         assert errornorm(Function(W).assign(10), u_reduce) < 1e-4
 
+    # check that u_reduce dat vector is still synchronised
+    ensemble_rank = ensemble.ensemble_comm.rank
+    spatial_rank = ensemble.comm.rank
+    if ensemble_rank == root:
+        from numpy import zeros
+        states = zeros(ensemble.comm.size, dtype=int)
+        with u_reduce.dat.vec as v:
+            states[spatial_rank] = v.stateGet()
+        ensemble.comm.Allgather(MPI.IN_PLACE, states)
+        for state in states:
+            assert state == states[0]
+
```

Comment:
Can you also check that the state on `root` is now _different_ from the state on the other ensembles?

Reply:
I'm happy to put this in, but isn't this testing the implementation not the interface?
We want the state of the vector to be the same across the comm it's defined on (the ensemble member). But whether it is the same as on another rank is not part of the contract.

Reply:
I think this is correct.